src/sentry/utils/avatar.py

```diff
@@ -53,6 +54,7 @@ def get_gravatar_url(email, size=None, default='mm'):
 
 
 def hash_user_identifier(identifier):
+    identifier = force_bytes(identifier, errors='replace')
     return sum(map(ord, str(identifier)))
```

Comment:
Just replace the str() entirely with the force_bytes call. force_bytes, as the name implies, returns bytes already.